net-core/Ical.Net/Ical.Net.UnitTests/DataTypes/RequestStatusTest.cs

```diff
@@ -1,122 +1,126 @@
-            object requestStatusB = new RequestStatus();
-            object requestStatusC = requestStatusA;
-            object requestStatusD = new RequestStatus()
-            {
-                Description = "ABC Description"
-            };
+            Assert.AreEqual(expected, incoming.Equals(other));
+        }
 
-            Assert.IsFalse(requestStatusA.Equals(null));
-            Assert.IsFalse(requestStatusA.Equals("Invalid type value"));
-            Assert.IsTrue(requestStatusA.Equals(requestStatusA));
-            Assert.IsTrue(requestStatusA.Equals(requestStatusB));
-            Assert.IsTrue(requestStatusA.Equals(requestStatusC));
-            Assert.IsFalse(requestStatusA.Equals(requestStatusD));
+        [Test, TestCaseSource(nameof(EqualsTestCases))]
+        public void EqualsTests(RequestStatus incoming, object other, bool expected)
+        {
+            Assert.AreEqual(expected, incoming.Equals((object)other));
         }
 
-        [Test]
-        public void GetHashCode_ValidObject()
+        private static IEnumerable<ITestCaseData> GetHashCodeTestCases()
         {
-            var requestStatusA = new RequestStatus();
-            var requestStatusB = new RequestStatus() 
+            var emptyRequestStatus = new RequestStatus();
+            yield return new TestCaseData(emptyRequestStatus, 0)
+                .SetName("GetHashCode() for empty request status");
+
+            var emptyPropertiesRequestStatus = new RequestStatus()
             {
                 Description = "",
                 ExtraData = "",
                 StatusCode = new StatusCode()
             };
-            var requestStatusC = new RequestStatus() 
+            yield return new TestCaseData(emptyPropertiesRequestStatus, 666477112)
+                .SetName("GetHashCode() for request status with empty properties");
+
+            var normalRequestStatus = new RequestStatus()
             {
                 Description = "ABC Description",
                 ExtraData = "ABC ExtraData",
                 StatusCode = new StatusCode()
             };
-
-            Assert.AreEqual(0,  requestStatusA.GetHashCode());
-            Assert.AreEqual(666477112, requestStatusB.GetHashCode());
-            Assert.AreEqual(-150101454, requestStatusC.GetHashCode());
+            yield return new TestCaseData(normalRequestStatus, -150101454)
+                .SetName("GetHashCode() for initialised request status");
         }
 
-        [Test]
-        public void CopyFrom_NullInput() 
+        [Test, TestCaseSource(nameof(GetHashCodeTestCases))]
+        public void GetHashCodeTests(RequestStatus incoming, int expected)
         {
-            var requestStatus = new RequestStatus();
-            requestStatus.CopyFrom(null);
-
-            Assert.IsNull(requestStatus.StatusCode);
-            Assert.IsNull(requestStatus.Description);
-            Assert.IsNull(requestStatus.ExtraData);
+            Assert.AreEqual(expected, incoming.GetHashCode());
```

Comment:
The primary value of testing hash codes is not to assert what the value should be, but rather to make sure the hash is stable and deterministic. To that end, I would prefer to do a side by side comparison of hash values, instead of hard-coding a specific hash value. You did this already with `Equals()`. In most respect, `GetHashCode` and `Equals` are two sides of the same coin, and are amenable to similar test strategies. Inside, the values each function considers should be the same.

Example: consider a more complicated object like `CalendarEvent`. What if we changed the `Summary` string's hashing to be culture-specific? That would break a test that uses a hard-coded constant, even though the change isn't wrong. But if we compare the output of the two `GetHashCode()` calls, they'd better match, otherwise our hash function is bad. 

```csharp
var a = new RequestStatus();
var b = new RequestStatus();
Assert.AreEqual(a.GetHashCode(), b.GetHashCode());
```